crates/uv-cache/src/removal.rs

```diff
@@ -35,36 +35,53 @@ impl Removal {
             Err(err) => return Err(err),
         };
 
-        let mut rm_file = |path: &Path, meta: Result<std::fs::Metadata, walkdir::Error>| {
-            if let Ok(meta) = meta {
-                self.total_bytes += meta.len();
-            }
-            remove_file(path)?;
-
-            Ok(())
-        };
-
         if !metadata.is_dir() {
             self.num_files += 1;
-            return rm_file(path, Ok(metadata));
+
+            // Remove the file.
+            self.total_bytes += metadata.len();
+            remove_file(path)?;
+
+            return Ok(());
         }
 
         for entry in walkdir::WalkDir::new(path).contents_first(true) {
+            // If we hit a directory that lacks read permissions, try to make it readable.
+            if let Err(ref err) = entry {
+                if err
+                    .io_error()
+                    .is_some_and(|err| err.kind() == io::ErrorKind::PermissionDenied)
+                {
+                    if let Some(dir) = err.path() {
+                        if set_readable(dir).unwrap_or(false) {
+                            // Retry the operation; if we _just_ `self.rm_rf(dir)` and continue,
+                            // `walkdir` may give us duplicate entries for the directory.
+                            return self.rm_rf(path);
```

Comment:
\cc @BurntSushi

Reply:
This is interesting. As in like, the error will be reported again in the iterator?